solr/core/src/test/org/apache/solr/update/SolrIndexSplitterTest.java

```diff
@@ -364,7 +364,104 @@ private void doTestSplitByRouteKey(SolrIndexSplitter.SplitMethod splitMethod) th
     }
   }
 
-  private List<DocRouter.Range> getRanges(String id1, String id2) throws UnsupportedEncodingException {
+  @Test
+  public void testSplitWithChildDocs() throws Exception {
+    doTestSplitWithChildDocs(SolrIndexSplitter.SplitMethod.REWRITE);
+  }
+
+  @Test
+  public void testSplitWithChildDocsLink() throws Exception {
+    doTestSplitWithChildDocs(SolrIndexSplitter.SplitMethod.LINK);
+  }
+
+  public void doTestSplitWithChildDocs(SolrIndexSplitter.SplitMethod splitMethod) throws Exception {
+    // Overall test/split pattern copied from doTestSplitByRouteKey
```

Comment:
Okay; has an appearance of an old test but I can sympathize being consistent with a similar test.